tests/api/test_parameters.py

```diff
@@ -172,7 +172,7 @@ def test_path_parameter_someint__bad(simple_app):
     # non-integer values will not match Flask route
     app_client = simple_app.app.test_client()
     resp = app_client.get("/v1.0/test-int-path/foo")  # type: flask.Response
-    assert resp.status_code == 400, resp.text
+    assert resp.status_code == 404, resp.text
```

Comment:
Due to the fix for the typed path params, we now return a 404 again instead of a 400 when sending the wrong type of path parameter. This is because the request is no longer matched by the router, so it doesn't hit the parameter validation. It is either rejected by the Middleware router, or directly passed to the App router, where it is rejected.